pkg/controller/gitrepo/gitrepo_reconcile.go

```diff
@@ -42,6 +42,9 @@ func (r *ReconcileGitRepo) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 	helpers.AddTenantLabel(&instance.ObjectMeta, instance.Spec.TenantRef.Name)
+	if instance.Spec.RepoType == synv1alpha1.DefaultRepoType {
```

Comment:
It's possible to set defaults via markers directly in the CRD struct: https://book.kubebuilder.io/reference/markers/crd-validation.html

So we don't have to clutter the logic with such if statements.

Reply:
Unfortunately this only seems to be GA since K8s v1.17: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting

I guess we should leave the code there for now?

Reply:
That's unfortunate but to support older clusters let's leave the code.